python/triton/compiler/compiler.py

```diff
@@ -276,6 +276,12 @@ def compile(src, target=None, options=None):
             print(f"\nOverriding kernel with file {ir_filename}")
             full_name = fn_override_manager.get_file(ir_filename)
             next_module = parse(full_name, ext, context)
+        # use an env variable to parse ttgir from file
+        use_ttgir_loc = os.environ.get("USE_TTGIR_LOC", "0") == "1"
+        if use_ttgir_loc and ext == "ttgir":
+            ttgir_full_name = fn_cache_manager.get_file(ir_filename)
+            next_module = parse(ttgir_full_name, ext, context)
```

Comment:
Why parse it again if `next_module` is already a ttgir module?

Reply:
We need to parse it again to get the line numbers of the ttgir file :] When there is no #loc in the ttgir file, parsing it again will set the location of ops in the ttgir to the line number in the ttgir file.

Reply:
I get it now. Thanks.

I think the following behavior might make more sense as I described above.

```
DISABLE_LOC_DUMP = 0, USE_TTGIR_LOC = 1, ttgir loc info.
DISABLE_LOC_DUMP = 1, USE_TTGIR_LOC = 1, no loc info.
DISABLE_LOC_DUMP = 0, USE_TTGIR_LOC = 0, python loc info.
DISABLE_LOC_DUMP = 1, USE_TTGIR_LOC = 0, no loc info.
```

Reply:
> Does it make sense to only use `TRITON_DISABLE_LINE_INFO` but not `DISABLE_LOC_DUMP`? I'm a bit worried about the growth of env flags.

If we set TRITON_DISABLE_LINE_INFO to 1, there will be no debug info/location info. I wonder if it makes sense to get rid of DISABLE_LOC_DUMP and only use USE_TTGIR_LOC, when USE_TTGIR_LOC is true, we will not dump location info in the cached ttgir so when we parse it again, it will use the ttgir line number.

Reply:
Sure, that seems better to me